dinky-admin/src/main/java/org/dinky/service/impl/UserServiceImpl.java

```diff
@@ -221,7 +221,17 @@ private void insertToken(UserDTO userInfo) {
         sysToken.setCreator(userId);
         sysToken.setUpdater(userId);
         sysToken.setSource(SysToken.Source.LOGIN);
-        tokenMapper.insert(sysToken);
+
+        synchronized (this) {
+            SysToken lastSysToken =
```

Comment:
Will there be a problem locking the entire class instance?

Reply:
ok,i will  use reentranlock to avoid this

Reply:
The read and write operation is a two-step process, and locking is used to avoid unique index conflicts during writing caused by read concurrency in extreme cases

Reply:
Indeed, it is